applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_mgmt_scan.h"
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/hci.h>
+
+#include "ble_hci_vsc.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_mgmt_scan, CONFIG_BLE_LOG_LEVEL);
+
+#define CONNECTION_PARAMETERS                                                                      \
+	BT_LE_CONN_PARAM(CONFIG_BLE_ACL_CONN_INTERVAL, CONFIG_BLE_ACL_CONN_INTERVAL,               \
+			 CONFIG_BLE_ACL_SLAVE_LATENCY, CONFIG_BLE_ACL_SUP_TIMEOUT)
+
+static bt_mgmt_conn_set_cb conn_set;
+static bt_mgmt_conn_disconnected_cb conn_disconnected;
+
+static int scan_interval_ms;
+static int scan_window_ms;
+
+static uint8_t num_con;
```

Comment:
```suggestion
static uint8_t num_conn;
```
Can this be gotten from host directly?

Reply:
If not, maybe we should create our own using bt_conn_foreach combined with bt_conn_get_info. Keeping a static counter in our code is asking for bugs.

Reply:
Good point.
Changed to using bt_conn_foreach and bt_conn_get_info, and removed the static counter.
This is the best way available with the current host API,